Formik login signup

NasimFatima opened the pull request.

NasimFatima committed:
using semantic ui react component

NasimFatima committed:
login, signup, authentication, login with google, redux setup

NasimFatima committed:
material ui react

NasimFatima committed:
remove semantic ui component

NasimFatima committed:
remove app.js file

NasimFatima committed:
remove commented code

NasimFatima committed:
using formik for forms

mareenanazir commented on src/Services/axios.js:
```diff
@@ -0,0 +1,31 @@
+
+const baseURL = process.env.REACT_APP_BACKEND_URL;
+let headers = {};
+// if (localStorage.token) {
```
remove unused code.

mareenanazir commented on src/Services/axios.js:
```diff
@@ -0,0 +1,31 @@
+  return response;
+}, (error) => {
+  if (error.response.status === 401) {
+    localStorage.removeItem("token")
```
why are using localStorage?

mareenanazir commented on src/components/GoogleLogin/GoogleLogin.js:
```diff
@@ -0,0 +1,38 @@
+  const history = useHistory()
+  const dispatch = useDispatch();
+  const successResponseGoogle = response => {
+    API.post(`users/google/`, { token: response.accessToken })
```
we need to make a generic APICaller and use that everywhere instead of calling the axios.

mareenanazir commented on src/components/GoogleLogin/GoogleLogin.js:
```diff
@@ -0,0 +1,38 @@
+  const dispatch = useDispatch();
+  const successResponseGoogle = response => {
+    API.post(`users/google/`, { token: response.accessToken })
+      .then(res => {
```
all the endpoints should be in a separate file rather than in each component

mareenanazir commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,26 @@
+
+export const Headers = () => {
+  return (
+    <div>
```
why are we using these extra divs?

mareenanazir commented on src/components/Header/Header.js:
```diff
@@ -0,0 +1,26 @@
+  return (
+    <div>
+      <div>
+        <AppBar position="static">
```
be consistent with single or double quotes

mareenanazir commented on src/components/Home/Home.js:
```diff
@@ -0,0 +1,21 @@
+import API from '../../Services/axios';
+import { Headers } from '../Header'
+export const Home = () => {
+  API.get(`users/dummy/`)
```
why are we doing async actions directly in the component? Why not using useEffect?

mareenanazir commented on src/components/Login/Login.js:
```diff
@@ -0,0 +1,90 @@
+        <h1>Login to Your Account</h1>
+        <FormInput>
+          <label htmlFor="email">Email</label>
+          <input
```
use Field from formik instead of input

mareenanazir commented on src/components/Login/Login.js:
```diff
@@ -0,0 +1,90 @@
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+          />
+          {formik.touched.email && formik.errors.email ? (
```
use && operator instead

mareenanazir commented on src/components/Signup/Signup.js:
```diff
@@ -0,0 +1,205 @@
+/* eslint-disable no-unused-vars */
+import React, { useState, useEffect } from 'react';
+// import Button from '@material-ui/core/Button';
```
remove the comments

mareenanazir commented on src/components/Signup/Signup.js:
```diff
@@ -0,0 +1,205 @@
+  const dispatch = useDispatch();
+
+  // const [signupForm, setSignupForm] = useState({});
+  const [roles, setRoles] = useState([]);
```
we should not use local state when using redux

mareenanazir commented on src/components/Signup/Signup.js:
```diff
@@ -0,0 +1,205 @@
+  const formik = useFormik({
+    initialValues: { first_name: '', last_name: '', email: '', password1: '', password2: '', phone: '', address: '', role: -1 },
+    validationSchema: Yup.object({
+      first_name: Yup.string()
```
use camelcase not snake case

mareenanazir commented on src/redux/Reducers/UserAuthReducer.js:
```diff
@@ -0,0 +1,26 @@
+
+const initial_state = {
+  user: {
+    pk: 0,
```
user is fine, we don't need to define the inner keys in initial state

mareenanazir commented on src/routes/routesMapping.js:
```diff
@@ -0,0 +1,34 @@
+import { routes } from './index'
+
+
+const RenderRoute = (route) => {
```
We should make a component named PrivateRoute and use that

mareenanazir reviewed (changes requested):
Use Sagas for async actions, Routing needs to be done by making separate components for PrivateRoute, also have a look at redux-persist.

NasimFatima committed:
Implement redux saga and changes after PR review

NasimFatima committed:
Menu and user add, view, sidebar, tables implementation

NasimFatima committed:
remove local state

NasimFatima committed:
home routes

NasimFatima committed:
add toastify, menu fixes, return response in camel case axios, signup and login links

NasimFatima committed:
logout implementation

NasimFatima committed:
Persistent Storage
Order Placement
Add to cart
Checkout
Role based Access
Application flow
Css changes

NasimFatima committed:
Customers should be able to view items stored in the cart, on any device that they login in into.
They should be able to mark an order as complete

NasimFatima committed:
style components and ui fixes

NasimFatima committed:
fixe issues reflected by backend changes
